### PR TITLE
Refactor tests and benchmarks

### PR DESCRIPTION
**Pull request by AleCandido**

### CODE REVIEW

**Comment by AleCandido** [comment]
@AlexRaiteri I removed the whole `test_vs_rgsl/` because it contained the old tests in C, while in #9 you're already writing the benchmarks in Rust with Criterion (nice choice, thanks for introducing it :+1:), so these should be outdated.

If you can take a look at the other changes, then I would merge this PR, and keep going with your benchmarks :)

File: Cargo.toml [hunk 1]
```diff
@@ -7,32 +7,22 @@ license = "GPL-3.0-or-later"
...
-# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
-
 [lib]
 name = "quad"
-# "cdylib" is necessary to produce a shared library for Python to import from.
-#
-# Downstream Rust code (including code in `bin/`, `examples/`, and `tests/`) will not be able
-# to `use python_bidings;` unless the "rlib" or "lib" crate type is also included, e.g.:
-# crate-type = ["cdylib", "rlib"]
 crate-type = ["cdylib"]
```
Comment (AlexRaiteri) [inline Cargo.toml:12]: bench seems to work with "["cdylib", "rlib"]" crate-type

File: Cargo.toml [hunk 2]
```diff
@@ -7,32 +7,22 @@ license = "GPL-3.0-or-later"
...
 
-
-[profile.dev]
-opt-level = 3
-
-[profile.release]
-opt-level = 3
\ No newline at end of file
+[dev-dependencies]
+GSL = "6.0.0"
```
[the remark this answers is not in the document]
Comment (AlexRaiteri) [reply Cargo.toml:22]: @AleCandido

File: Cargo.toml (hunk 1 above)
Comment (AlexRaiteri) [reply Cargo.toml:12]: @AleCandido

Comment (AleCandido) [reply Cargo.toml:12]:
```suggestion
crate-type = ["cdylib", "rlib"]
```

Just to show you how a suggestion looks like :)

**Pull request merged**